sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -287,8 +287,15 @@ def to_arrow(self) -> pyarrow.Table:
 
 
 def block_until_done(client, bq_job):
+    def _cancel_job(job_id):
+        client.cancel_job(job_id)
+        raise BigQueryJobCancelled(job_id=job_id)
+
     def _is_done(job_id):
-        return client.get_job(job_id).state in ["PENDING", "RUNNING"]
+        try:
+            return client.get_job(job_id).state in ["DONE", "SUCCESS"]
```

Comment:
@MattDelac Is this right? I'm not too familiar with the retry logic, but currently the job will be cancelled despite succeeding via the code a few lines below:

```python
    if not _is_done(job_id):
        _cancel_job(job_id)
```

Reply:
Before, it seems like we might be giving back control to the user before the job is finished, which could be a problem if they are trying to access the results (i.e. to be stored in a dest. table) right?